### ISSUE TITLE
throw expression in ref returning methods produces an error

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#16171]
**Version Used**: https://github.com/dotnet/roslyn/commit/a2df65047264ed08fb3a985056085dbcee821253

**Steps to Reproduce**:

```cs
class C
{
    ref int M() => throw null;
}
```
**Expected Behavior**: No error

**Actual Behavior**: `By-value returns may only be used in methods that return by value`
